docs/en/sql-reference/00-sql-reference/10-data-types/43-data-type-variant.md

```diff
@@ -139,22 +139,62 @@ VALUES
 Retrieve the preferred color for each user:
 ```sql
 SELECT
-  preferences['color'],
-  preferences:color,
-  profile['name'],
-  profile:name
+  preferences['settings']['color'],
+  preferences['settings']:color,
+  preferences['settings'].color,
+  preferences:settings['color'],
+  preferences:settings:color,
+  preferences:settings.color
 FROM
   user_preferences;
 ```
 
 Result:
 ```
-┌────────────────────────────────────────────────────────────────────────────────┐
-│ preferences['color'] │ preferences:color │  profile['name'] │   profile:name   │
-├──────────────────────┼───────────────────┼──────────────────┼──────────────────┤
-│ "red"                │ "red"             │ Amy              │ Amy              │
-│ "blue"               │ "blue"            │ Bob              │ Bob              │
-└────────────────────────────────────────────────────────────────────────────────┘
+┌────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ preferences['settings']['color'] │ preferences['settings']:color │ preferences['settings']:color │ preferences:settings['color'] │ preferences:settings:color │ preferences:settings:color │
+├──────────────────────────────────┼───────────────────────────────┼───────────────────────────────┼───────────────────────────────┼────────────────────────────┼────────────────────────────┤
+│ "red"                            │ "red"                         │ "red"                         │ "red"                         │ "red"                      │ "red"                      │
+│ "blue"                           │ "blue"                        │ "blue"                        │ "blue"                        │ "blue"                     │ "blue"                     │
+└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+```
+
+Please note that field names are **case-sensitive**. If a field name contains spaces or special characters, enclose it in double quotes.
+
+```sql
+INSERT INTO
+  user_preferences
+VALUES
+  (
+    3,
+    parse_json('{"new settings":{"color":"red", "fontSize":16, "theme":"dark"}}'),
```

Comment:
`parse_json` function is not needed, we can auto-parse JSON string to variant when inserting values.